contracts/hackatom/tests/integration.rs

```diff
@@ -339,8 +339,9 @@ mod singlepass_tests {
         // Gas consumtion is relatively small
         // Note: the exact gas usage depends on the Rust version used to compile WASM,
         // which we only fix when using cosmwasm-opt, not integration tests.
-        assert!(gas_used > 26000, "{}", gas_used);
-        assert!(gas_used < 30000, "{}", gas_used);
+        let expected = 42000; // +/- 20%
```

Comment:
Wow, this increased gas usage a lot! (+/- 28k to +/- 42k)

Maybe it makes sense to have a hot-path when there is no escaped characters inside the string to just do the old copy. We can merge this, but it would be nice to update the json lib before 0.8 (we can do that in a week or so, but it should be possible to do around the same cost when there are no escapes)